Makefile

```diff
@@ -1,43 +1,59 @@
+.SECONDEXPANSION:
+PERCENT := %
+
 CXX := clang++
 
-VERSION_HEADER := base/version.generated.h
+# TODO(egg): build benchmarks
+
+PLUGIN_TRANSLATION_UNITS       := $(wildcard ksp_plugin/*.cpp)
+PLUGIN_TEST_TRANSLATION_UNITS  := $(wildcard ksp_plugin_test/*.cpp)
+JOURNAL_TRANSLATION_UNITS      := $(wildcard journal/*.cpp)
+MOCK_TRANSLATION_UNITS         := $(wildcard */mock_*.cpp)
+BENCHMARK_TRANSLATION_UNITS    := $(wildcard benchmarks/*.cpp)
+TEST_TRANSLATION_UNITS         := $(wildcard */*_test.cpp)
+TEST_OR_MOCK_TRANSLATION_UNITS := $(TEST_TRANSLATION_UNITS) $(MOCK_TRANSLATION_UNITS)
+TOOLS_TRANSLATION_UNITS        := $(wildcard tools/*.cpp)
+LIBRARY_TRANSLATION_UNITS      := $(filter-out $(TEST_OR_MOCK_TRANSLATION_UNITS) $(BENCHMARK_TRANSLATION_UNITS), $(wildcard */*.cpp))
+JOURNAL_LIB_TRANSLATION_UNITS  := $(filter-out $(TEST_OR_MOCK_TRANSLATION_UNITS), $(wildcard journal/*.cpp))
+BASE_LIB_TRANSLATION_UNITS     := $(filter-out $(TEST_OR_MOCK_TRANSLATION_UNITS), $(wildcard base/*.cpp))
+PROTO_FILES                    := $(wildcard */*.proto)
+PROTO_TRANSLATION_UNITS        := $(PROTO_FILES:.proto=.pb.cc)
+PROTO_HEADERS                  := $(PROTO_FILES:.proto=.pb.h)
 
-CPP_SOURCES := ksp_plugin/plugin.cpp ksp_plugin/interface.cpp ksp_plugin/physics_bubble.cpp 
-TOOLS_SOURCES := $(wildcard tools/*.cpp)
-PROTO_SOURCES := $(wildcard */*.proto)
-PROTO_CC_SOURCES := $(PROTO_SOURCES:.proto=.pb.cc)
-PROTO_HEADERS := $(PROTO_SOURCES:.proto=.pb.h)
+DEP_DIR := deps/
 
-GENERATED_SOURCES := journal/profiles.generated.h \
-	journal/profiles.generated.cc \
-	journal/player.generated.cc \
+OBJ_DIRECTORY := obj/
+
+BIN_DIRECTORY := bin/
+TOOLS_BIN     := $(BIN_DIRECTORY)tools
+
+GMOCK_TRANSLATION_UNITS :=                     \
```

Comment:
oh. It's tab-indented because make. Let's put a single space before the first `\` and align the others.